### ISSUE TITLE
Ability to define runtime/non-runtime files and clean respectively

### ISSUE THREAD

**Issue by estahn** [composer/composer#7882]
TL;DR: Setting in composer to define non-required files and command to purge those.

In our world of Kubernetes and docker images, we try to keep images as small as possible. While disk space is usually not an issue we keep image sizes small for fast scalability/quick deployments due to data transfer between nodes/registry, IO limitations, etc.

There are [some libraries](https://packagist.org/?query=composer%20cleanup) that attempt to solve this problem by removing files based on "best practises" (e.g. `<root>/tests`). This probably works most of the time but fails in certain edge-cases or doesn't catch all files that could be removed.

A feature to define and remove these non-required files would be very useful, e.g.

```composer.json
{
  "runtime" : ["src/"]
}
```

```bash
composer clear-runtime
```

**Comment by alcohol** [comment]
This is not a concern that Composer should deal with. Users can solve this in many ways already.

**Comment by alcohol** [comment]
This topic has been discussed multiple times already. There simply is not enough to gain for the effort it would require to build this and have it adopted by the community. The impact is also completely negligible. Even for docker image layers, the impact these files have on layer size is negligible. Layers are stored and send over network transports in a compressed format, and text compresses really well.
